src/job/enhancedJobManager.ts

```diff
@@ -0,0 +1,53 @@
+import { IConfig } from 'config';
+import { JobRepository } from '../repositories/jobRepository';
+import { RepoBranchesRepository } from '../repositories/repoBranchesRepository';
+import { ICDNConnector } from '../services/cdn';
+import { IJobCommandExecutor } from '../services/commandExecutor';
+import { IFileSystemServices } from '../services/fileServices';
+import { IJobRepoLogger } from '../services/logger';
+import { IRepoConnector } from '../services/repo';
+import { JobHandlerFactory, JobManager } from './jobManager';
+import { IJobValidator } from './jobValidator';
+
+export class EnhancedJobManager extends JobManager {
+  private _cleanUp: () => Promise<void>;
+  constructor(
+    config: IConfig,
+    jobValidator: IJobValidator,
+    jobHandlerFactory: JobHandlerFactory,
+    jobCommandExecutor: IJobCommandExecutor,
+    jobRepository: JobRepository,
+    cdnConnector: ICDNConnector,
+    repoConnector: IRepoConnector,
+    fileSystemServices: IFileSystemServices,
+    logger: IJobRepoLogger,
+    repoBranchesRepo: RepoBranchesRepository,
+    cleanUp: () => Promise<void>
+  ) {
+    super(
+      config,
+      jobValidator,
+      jobHandlerFactory,
+      jobCommandExecutor,
+      jobRepository,
+      cdnConnector,
+      repoConnector,
+      fileSystemServices,
+      logger,
+      repoBranchesRepo
+    );
+    this._cleanUp = cleanUp;
+  }
+
+  override async work() {
+    while (true) {
+      const job = await this.getQueuedJob();
```

Comment:
[getQueuedJob](https://github.com/mongodb/docs-worker-pool/blob/303798fec002e9286bf1bb7d6b482dbf15a78c81/src/job/jobManager.ts#L131) seems to query MongoDB and uses it as its "queue" instead of polling for messages from the SQS. Is this intentional?

Reply:
Ah, good catch. I made an assumption from the `app.ts` code that this is reading from the SQS queue, but you're right, it's not. We can discuss in standup or later about how we want to go about this. It might be a smaller change to follow this pattern as opposed to having the task listening to the queue.